docs/source/en/model_doc/greaselm.mdx

```diff
@@ -0,0 +1,114 @@
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# greaselm
+
+## Overview
+
+The GreaseLM model was proposed in [GreaseLM: Graph REASoning Enhanced Language Models for Question Answering](https://arxiv.org/abs/2201.08860) by Xikun Zhang, Antoine Bosselut, Michihiro Yasunaga, Hongyu Ren, Percy Liang, Christopher D. Manning, Jure Leskovec.
+GreaseLM enables the fusion and exchange of information from the language model and the knowledge graph (KG) in multiple
+layers of its architecture. GreaseLM takes as input the natural language context and a graph neural network that reasons
+over the KG, ultimately demonstrating superior results in commonsense reasoning.
+
+The abstract from the paper is the following:
+
+Answering complex questions about textual narratives requires reasoning over both stated context and the world
+knowledge that underlies it. However, pretrained language models (LM), the foundation of most modern QA systems,
+do not robustly represent latent relationships between concepts, which is necessary for reasoning. While knowledge
+graphs (KG) are often used to augment LMs with structured representations of world knowledge, it remains an open
+question how to effectively fuse and reason over the KG representations and the language context, which provides
+situational constraints and nuances. In this work, we propose GreaseLM, a new model that fuses encoded representations
+from pretrained LMs and graph neural networks over multiple layers of modality interaction operations. Information
+from both modalities propagates to the other, allowing language context representations to be grounded by structured
+world knowledge, and allowing linguistic nuances (e.g., negation, hedging) in the context to inform the graph
+representations of knowledge. Our results on three benchmarks in the commonsense reasoning (i.e., CommonsenseQA,
+OpenbookQA) and medical question answering (i.e., MedQA-USMLE) domains demonstrate that GreaseLM can more reliably
+answer questions that require reasoning over both situational constraints and structured knowledge, even outperforming
+models 8x larger.
+
+Tips:
+
+```python
+import torch
+from transformers import GreaseLMProcessor, GreaseLMForMultipleChoice
+
+device = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+def index_to_answer(index, choices=["A", "B", "C", "D", "E"]):
+    return choices[index]
+
+
+proc = GreaseLMProcessor.from_pretrained("Xikun/greaselm-csqa")
+
+model = GreaseLMForMultipleChoice.from_pretrained("Xikun/greaselm-csqa")
+_ = model.to(device)
+_ = model.eval()
+
+csqa = {
```

Comment:
csqa probably means "common sense question-answer pair"?

Would advocate to use more comprehensive variable names.